src/App.tsx

```diff
@@ -15,36 +20,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const goodsWithIds: GoodType[] = goodsFromServer.map((good, index) => {
+  return {
+    name: good,
+    id: index,
+  };
+});
+
+enum SortType {
+  Default = '',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+function getPreparedGoods(
+  goods: GoodType[],
+  { sortField }: { sortField: SortType },
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return good1.name.localeCompare(good2.name);
+        case SortType.Length:
+          return good1.name.length - good2.name.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.Default);
+  const [reversed, setReversed] = useState(false);
+
+  let visibleGoods = getPreparedGoods(goodsWithIds, { sortField });
+
+  if (reversed) {
+    visibleGoods = visibleGoods.toReversed();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortType.Alphabet,
+          })}
+          onClick={() => {
+            setSortField(SortType.Alphabet);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => {
+            setSortField(SortType.Length);
+          }}
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SortType.Length,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setReversed(!reversed)}
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !reversed,
+          })}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {sortField || reversed ? (
+          <button
+            onClick={() => {
+              setSortField(SortType.Default);
+              setReversed(false);
+            }}
+            type="button"
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        ) : null}
```

Comment:
```suggestion
        {sortField || reversed && (
          <button
            onClick={() => {
              setSortField(SortType.Default);
              setReversed(false);
            }}
            type="button"
            className="button is-danger is-light"
          >
            Reset
          </button>
        )}
```

In this case you can use `&&` instead of `? :`